### ISSUE TITLE
Lab 4 Question on orElse outputs

### ISSUE THREAD

**Issue by AndersonDjumin** [nus-cs2030/2324-s2#271]
I am trying to use escape keys to print out the desired output but the backslashes show. I am told if I do SOUT for this string the backslashes should not show but in jshell it shows for some reason.

.map(x -> x.getGrade())
            .orElse("\"No Such Record: " + std + " " + course + " " + asm + "\"");

Does code crunch also output as such?

**Comment by tarshalim** [comment]
i think you just need to write like this and the backslashes will automatically show:
 .orElse("No such record: "
                      + std+ " "
                      + course+ " "
                      + asm);

**Comment by ezclap02** [comment]
yes, fully agree with @zuoshihua! string formatting makes printing outputs much easier

**Comment by benjaminseoww** [comment]
I completely agree with @zuoshihua about the use of String.format 

An even foolproof way is to assign the String output on a separate line so that you can visualise it better.

String output = "No Such Record: " + std + " " + course + " " + asm + "";
.map(x -> x.getGrade()).orElse(output);

**Comment by hanlinnnnn** [comment]
much more convenient to learn how to use String.format with %s %f %d etc etc, its the same logic as our "".format() in python last sem, or if ure more familiar with c then this is sprintf if im not wrong

**Comment by zay-nin** [comment]
agree with @zuoshihua ! it might be quite confusing at first, but some of the main ones we use would be %s for string, %d for integer, and %._f for doubles (where _ is the number of decimal places you would like in your output string)! hope this helps :)

**Comment by cherylkohzw** [comment]
I think you can do it this way, directly converting optional object to string

Optional<String> opt = super.get(name)
                .flatMap(x -> x.get(course))
                .flatMap(x -> x.get(ass))
                .map(x -> x.getGrade());

        String orElseOutput = String.format("No such record: %s %s %s",
                name, course, ass);
        return opt.orElse(orElseOutput);

**Comment by ZHD1987E** [comment]
@AndersonDjumin

Mind if you close this issue to declutter the 'issues' section?

Thanks.

**Comment by AndersonDjumin** [comment]
Issue closed @ZHD1987E thanks for the bump
